base/io.jl

```diff
@@ -152,15 +155,35 @@ read(stream, t)
     write(filename::AbstractString, x)
 
 Write the canonical binary representation of a value to the given I/O stream or file.
-Return the number of bytes written into the stream.   See also [`print`](@ref) to
+Return the number of bytes written into the stream. See also [`print`](@ref) to
 write a text representation (with an encoding that may depend upon `io`).
 
+The endianness of the written value depends on the endianness of the host system.
+Convert to/from a fixed endianness when writing/reading (e.g. using  [`htol`](@ref) and
+[`ltoh`](@ref)) to get results that are consistent across plattforms.
+
 You can write multiple values with the same `write` call. i.e. the following are equivalent:
 
     write(io, x, y...)
     write(io, x) + write(io, y...)
 
 # Examples
+Consistent serialization:
+```jldoctest
+julia> open("/tmp/test.bin","w") do f
```

Comment:
Perhaps use `mktempdir()` instead to be more portable?

Reply:
Does it matter? It's only an example. And everybody should understand file paths, while me for one had to look up first what `tempname()` does.

Reply:
Well, it is not just an example, it is a doctest and we should try our best to make sure that these are portable. I can push a commit here if you don't feel like doing it.

Reply:
I'm working on it. Is there a way to tell the doctest that some output will change between runs? I would show the output of `tempname()` then, as this allows people to immediately see what `fname = tempname()` really is.

Reply:
You can just add a comment instead:
```
julia> fname = tempname(); # random temporary file name
```

Reply:
Great, thanks.